api/mongowriter.go

```diff
@@ -49,3 +86,35 @@ func (mw *MongoWriter) WriteHeader(statusCode int) {
 	}
 	mw.w.WriteHeader(statusCode)
 }
+
+// ErrorBuffer returns the data stored in the error buffer.
+func (mw *MongoWriter) ErrorBuffer() []byte {
+	return mw.ew.Buffer.Bytes()
+}
+
+// ErrorStatus returns the status code with which the last call errored out,
+// if any.
+func (mw *MongoWriter) ErrorStatus() int {
+	return mw.ew.Status
+}
+
+// FailedWithWriteConflict informs us whether the MongoWriter received a MongoDB
+// WriteConflict error.
+func (mw *MongoWriter) FailedWithWriteConflict() bool {
```

Comment:
We should write a basic unit test that hits these methods.

Reply:
Added.